sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/PhoneNumbersAsyncClient.java

```diff
@@ -0,0 +1,377 @@
+
+import com.azure.communication.phonenumbers.implementation.PhoneNumberAdminClientImpl;
+import com.azure.communication.phonenumbers.implementation.PhoneNumbersImpl;
+import com.azure.communication.phonenumbers.implementation.models.PhoneNumbersPurchasePhoneNumbersResponse;
+import com.azure.communication.phonenumbers.implementation.models.PhoneNumberPurchaseRequest;
+import com.azure.communication.phonenumbers.implementation.models.PhoneNumbersSearchAvailablePhoneNumbersResponse;
+import com.azure.communication.phonenumbers.implementation.models.PhoneNumbersReleasePhoneNumberResponse;
+import com.azure.communication.phonenumbers.implementation.models.PhoneNumbersUpdateCapabilitiesResponse;
+import com.azure.communication.phonenumbers.models.AcquiredPhoneNumber;
+import com.azure.communication.phonenumbers.models.PhoneNumberCapabilitiesRequest;
+import com.azure.communication.phonenumbers.models.PhoneNumberOperation;
+import com.azure.communication.phonenumbers.models.PhoneNumberOperationStatus;
+import com.azure.communication.phonenumbers.models.PhoneNumberSearchRequest;
+import com.azure.communication.phonenumbers.models.PhoneNumberSearchResult;
+import com.azure.communication.phonenumbers.models.PhoneNumberUpdateRequest;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.http.rest.PagedFlux;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.Context;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.polling.LongRunningOperationStatus;
+import com.azure.core.util.polling.PollResponse;
+import com.azure.core.util.polling.PollerFlux;
+import com.azure.core.util.polling.PollingContext;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.FluxUtil.withContext;
+
+/**
+ * Asynchronous client for Communication service phone number operations
+ */
+@ServiceClient(builder = PhoneNumbersClientBuilder.class, isAsync = true)
+public final class PhoneNumbersAsyncClient {
+    private final ClientLogger logger = new ClientLogger(PhoneNumbersAsyncClient.class);
+    private final PhoneNumbersImpl client;
+    private final Duration defaultPollInterval = Duration.ofSeconds(1);
+
+    PhoneNumbersAsyncClient(PhoneNumberAdminClientImpl phoneNumberAdminClient) {
+        this.client = phoneNumberAdminClient.getPhoneNumbers();
+    }
+
+    /**
+     * Gets information about an acquired phone number.
+     * @param phoneNumber The phone number id in E.164 format. The leading plus can be either + or encoded
+     *                    as %2B.
+     * @return {@link AcquiredPhoneNumber} representing the acquired telephone number.
```

Comment:
@throws: We normally document runtime exception. For example in this case NullPointerException .  This apply to all the API below.

Reply:
Fixed thank you